### PR TITLE
feat: allow storing the GitHub token in `.GITHUB_TOKEN` file

### PR DESCRIPTION
**Pull request by sockmaster27**
Fixes: #6972

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/Main.scala [hunk 1]
```diff
@@ -69,10 +71,22 @@ object Main {
...
     }
 
-    // Get GitHub token
+    // get GitHub token
     val githubToken =
       cmdOpts.githubToken
+        .orElse(FileOps.readString(cwd.resolve("./.GITHUB_TOKEN")))
         .orElse(sys.env.get("GITHUB_TOKEN"))
+        .map { t =>
+          val trimmed = t.trim
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/Main.scala:80]: This code here breaks the flow and seem unnatural. May I instead suggest to have FileOps.readLine which reads only the first line?

File: main/src/ca/uwaterloo/flix/api/Bootstrap.scala [hunk 2]
```diff
@@ -91,6 +91,7 @@ object Bootstrap {
          |artifact/
          |build/
          |lib/
+         |.GITHUB_TOKEN
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/Bootstrap.scala:94]: move up before the folders.

File: main/src/ca/uwaterloo/flix/util/FileOps.scala [hunk 3]
```diff
@@ -23,6 +23,22 @@ import java.nio.file.{Files, LinkOption, Path}
...
 
+  /**
+    * Reads the contents of the file at the given path `p` if it is possible.
+    *
+    * Returns None if:
+    * - The path doesn't exist.
+    * - The path isn't a file.
+    * - The current process doesn't have the necessary permissions.
+    */
+  def readString(p: Path): Option[String] = {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/util/FileOps.scala:34]: Maybe one line? Maybe also check its: ordinary file?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/util/FileOps.scala:34]: "regular file"

File: main/src/ca/uwaterloo/flix/Main.scala (hunk 1 above)
Comment (sockmaster27) [reply main/src/ca/uwaterloo/flix/Main.scala:80]: I agree, but secretly only reading the first line sounds a bit unpredictable. What about moving this part to a helper function called something like `sanitizeGithubToken`?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/Main.scala:80]: I think simple-stupid is best. What's wrong with reading just the first line?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/Main.scala:80]: I really dont think we want to start doing input validation on the token.

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**